tests/spellchecking.js

```diff
@@ -0,0 +1,259 @@
+		} );
+
+		it( 'should replace with shorter word (merging letter after)', () => {
+			emulateSpellcheckerMutation( editor, 0, 29,
+				'The Foo hous a is a Foo hous e. A Foo athat and Foo xhat. This is an istane',
+				'The Foo hous a is a Foo house. A Foo athat and Foo xhat. This is an istane' );
+
+			expectContent( editor,
+				'<paragraph>The Foo hous a is a Foo house[]. A Foo athat and Foo xhat. This is an istane</paragraph>' +
+				'<paragraph>Banana, orenge, appfle and the new comppputer</paragraph>',
+
+				'<p>The Foo hous a is a Foo house{}. A Foo athat and Foo xhat. This is an istane</p>' +
+				'<p>Banana, orenge, appfle and the new comppputer</p>' );
+		} );
+
+		it( 'should replace with same length text', () => {
+			emulateSpellcheckerMutation( editor, 0, 43,
+				'The Foo hous a is a Foo hous e. A Foo athat and Foo xhat. This is an istane',
+				'The Foo hous a is a Foo hous e. A Food that and Foo xhat. This is an istane' );
+
+			expectContent( editor,
+				'<paragraph>The Foo hous a is a Foo hous e. A Food that[] and Foo xhat. This is an istane</paragraph>' +
+				'<paragraph>Banana, orenge, appfle and the new comppputer</paragraph>',
+
+				'<p>The Foo hous a is a Foo hous e. A Food that{} and Foo xhat. This is an istane</p>' +
+				'<p>Banana, orenge, appfle and the new comppputer</p>' );
+		} );
+
+		it( 'should replace with longer word on the paragraph end', () => {
+			emulateSpellcheckerMutation( editor, 0, 77,
+				'The Foo hous a is a Foo hous e. A Foo athat and Foo xhat. This is an istane',
+				'The Foo hous a is a Foo hous e. A Foo athat and Foo xhat. This is an instance' );
+
+			expectContent( editor,
+				'<paragraph>The Foo hous a is a Foo hous e. A Foo athat and Foo xhat. This is an instance[]</paragraph>' +
+				'<paragraph>Banana, orenge, appfle and the new comppputer</paragraph>',
+
+				'<p>The Foo hous a is a Foo hous e. A Foo athat and Foo xhat. This is an instance{}</p>' +
+				'<p>Banana, orenge, appfle and the new comppputer</p>' );
+		} );
+
+		it( 'should replace with shorter word on the paragraph end', () => {
+			emulateSpellcheckerMutation( editor, 1, 43,
+				'Banana, orenge, appfle and the new comppputer',
+				'Banana, orenge, appfle and the new computer' );
+
+			expectContent( editor,
+				'<paragraph>The Foo hous a is a Foo hous e. A Foo athat and Foo xhat. This is an istane</paragraph>' +
+				'<paragraph>Banana, orenge, appfle and the new computer[]</paragraph>',
+
+				'<p>The Foo hous a is a Foo hous e. A Foo athat and Foo xhat. This is an istane</p>' +
+				'<p>Banana, orenge, appfle and the new computer{}</p>' );
+		} );
+	} );
+
+	describe( 'Plain text spellchecking (insertText)', () => {
+		// This tests emulates spellchecker correction on non-styled text by inserting correction text via native 'insertText' command.
+
+		it( 'should replace with longer word (collapsed)', ( done ) => {
+			emulateSpellcheckerInsertText( editor, 0, 12, 12, 'e' );
+
+			setTimeout( () => {
```

Comment:
Could we avoid using `setTimeout()`? Perhaps the check can be done on `editor.document#changeDone`? It'll be more reliable (if the timeout is longer). Also, it'd be good to know that the changes are done immediately.

PS. When using listener to check such a thing there may be issues with the event being fired after the test terminated. This will cause crashing tests in an unpredictable manner. So make sure to use a proper `afterEach()` with editor destroy. Also, you can secure it by using `once()`.